### PR TITLE
Deprecate `sum_expand` and `hamiltonian_expand`

### PR DESCRIPTION
**Pull request by Shiro-Raven**
[[sc-64018](https://app.shortcut.com/xanaduai/story/64018)]

### CODE REVIEW

File: pennylane/transforms/hamiltonian_expand.py [hunk 1]
```diff
@@ -223,6 +228,12 @@ def hamiltonian_expand(tape: QuantumTape, group: bool = True) -> (Sequence[Quant
     2
     """
 
+    warnings.warn(
+        "'qml.transforms.hamiltonian_expand' is deprecated and will be removed in version 0.39. "
```
Comment (trbromley) [inline pennylane/transforms/hamiltonian_expand.py:232]:
```suggestion
        "qml.transforms.hamiltonian_expand is deprecated and will be removed in version 0.39. "
```
We don't need (and don't normally do (?)) quotes/backticks in the code-based warnings.

File: doc/development/deprecations.rst [hunk 2]
```diff
@@ -9,6 +9,12 @@ deprecations are listed below.
 Pending deprecations
 --------------------
 
+* The functions ``qml.transforms.sum_expand`` and ``qml.transforms.hamiltonian_expand`` are deprecated.
+  Instead, ``qml.transforms.split_non_commuting`` can be used for an equivalent effect.
```
Comment (mudit2812) [inline doc/development/deprecations.rst:13]:
```suggestion
  Instead, ``qml.transforms.split_non_commuting`` can be used for equivalent behaviour.
```

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -49,6 +49,11 @@
 
 <h3>Deprecations 👋</h3>
 
+* `pennylane.transforms.sum_expand` and `pennylane.transforms.hamiltonian_expand` have been deprecated.
+  Users should instead use `pennylane.transforms.split_non_commuting` for an equivalent behaviour.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:82]:
```suggestion
  Users should instead use `pennylane.transforms.split_non_commuting` for equivalent behaviour.
```

File: pennylane/transforms/hamiltonian_expand.py (hunk 1 above)
Comment (mudit2812) [reply pennylane/transforms/hamiltonian_expand.py]: Same change for the line below this, and for `sum_expand` as well.

File: pennylane/transforms/hamiltonian_expand.py [hunk 4]
```diff
@@ -458,6 +473,12 @@ def sum_expand(tape: QuantumTape, group: bool = True) -> (Sequence[QuantumTape],
 
     """
 
+    warnings.warn(
+        "'qml.transforms.sum_expand' is deprecated and will be removed in version 0.39. "
+        "Instead, use 'qml.transforms.split_non_commuting', which can handle the same measurement type.",
```
Comment (mudit2812) [inline pennylane/transforms/hamiltonian_expand.py:478]:
```suggestion
        "qml.transforms.sum_expand is deprecated and will be removed in version 0.39. "
        "Instead, use qml.transforms.split_non_commuting, which can handle the same measurement type.",
```

File: pennylane/transforms/hamiltonian_expand.py [hunk 5]
```diff
@@ -223,6 +228,12 @@ def hamiltonian_expand(tape: QuantumTape, group: bool = True) -> (Sequence[Quant
     2
     """
 
+    warnings.warn(
+        "qml.transforms.hamiltonian_expand is deprecated and will be removed in version 0.39. "
+        "Instead, use 'qml.transforms.split_non_commuting', which can handle the same measurement type.",
```
Comment (mudit2812) [inline pennylane/transforms/hamiltonian_expand.py:233]:
```suggestion
        "Instead, use qml.transforms.split_non_commuting, which can handle the same measurement type.",
```

**Review by mudit2812 — APPROVED** [review]

**Review by PietropaoloFrisoni — APPROVED** [review]
Thanks!

**Pull request merged**